docs/v2-roadmap.rst

```diff
@@ -0,0 +1,179 @@
+v2.0 Roadmap
+============
+
+
+**📅 Release and Migration Schedule**
+-------------------------------------
+
+.. important::
+
+   We're seeking `sponsors and supporters for urllib3 v2.0 on Open Collective <https://opencollective.com/urllib3>`_.
+   There's a lot of work to be done for our small team and we want to make sure
+   development can get completed on-time while also fairly compensating contributors
+   for the additional effort required for a large release like ``v2.0``.
+
+   Additional information available within the :doc:`sponsors` section of our documentation.
+
+
+We're aiming for all ``v2.x`` features to be released in **mid-to-late 2021**.
+
+Here's what the release and migration schedule will look like leading up
+to v2.0 being released:
```

Comment:
I would move the migration schedule can be moved at the bottom, it's only relevant after users got excited about the new features

Reply:
But +1 about the contents of the migration guide itself, it is perfect!